### PR TITLE
[fix]: [CDS-77547]: Add new pipeline-execution start and end counter in our code flows with accountId and status

### PR DESCRIPTION
**Pull request by archit-harness**
## Describe your changes
Changes - Add new pipeline-execution start and end counter in our code flows with accountId and status

Impact - Publishing metrics to new timelines via code flow rather than iterator, should be no impact as there is no network call for sending these metrics, and are infact scraped by prometheus separately

Testing -
1. Ran few executions, did interrupts (abort), and tried both failed and success executions.

Screenshot 2023-08-22 at 6 48 10 PM

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/51917)

### CODE REVIEW

File: pipeline-service/service/src/main/resources/metrics/metricDefinitions/execution-starting-count-metrics.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+name: Count of Pipeline Executions Starting
```
Comment (Raghav-Murali) [inline pipeline-service/service/src/main/resources/metrics/metricDefinitions/execution-starting-count-metrics.yaml:1]: There is no need to have 2 separate files. We can have both the metrics in the same definition file.

File: pipeline-service/service/src/main/resources/metrics/metricGroups/execution-starting-count-metric-group.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+name: Group by fields for metrics
```
Comment (Raghav-Murali) [inline pipeline-service/service/src/main/resources/metrics/metricGroups/execution-starting-count-metric-group.yaml:1]: Same as previous comment. No need to have separate group file, one should suffice.

**Comment by archit-harness** [comment]
trigger uts

File: pipeline-service/service/src/main/resources/metrics/metricDefinitions/execution-starting-count-metrics.yaml (hunk 1 above)
Comment (archit-harness) [reply pipeline-service/service/src/main/resources/metrics/metricDefinitions/execution-starting-count-metrics.yaml:1]: @Raghav-Murali are you saying 2 files which i created for start and end? i want to have separate so wouldn't i need separate main file? group file maybe used common

**Comment by archit-harness** [comment]
trigger messagecheck

**Comment by archit-harness** [comment]
trigger uts

File: pipeline-service/service/src/main/java/io/harness/monitoring/PipelineMetricsPublisher.java
Comment (naidu-harness) [inline pipeline-service/service/src/main/java/io/harness/monitoring/PipelineMetricsPublisher.java:1]: cant we remove this file altogether?

**Review by naidu-harness — CHANGES_REQUESTED** [review]

File: pipeline-service/service/src/main/java/io/harness/pms/plan/execution/handlers/PipelineStatusUpdateEventHandler.java [hunk 3]
```diff
@@ -117,6 +123,10 @@ public void onEnd(Ambiance ambiance) {
                   pipelineExecutionSummaryUpdatedEntity.getEndTs()));
         }
       }
+
+      // Update pipeline execution metrics
```
Comment (hextechpal) [inline pipeline-service/service/src/main/java/io/harness/pms/plan/execution/handlers/PipelineStatusUpdateEventHandler.java:127]: again please write this in a diff observer

**Review by hextechpal — CHANGES_REQUESTED** [review]

File: pipeline-service/service/src/main/java/io/harness/pms/plan/execution/handlers/PipelineStatusUpdateEventHandler.java (hunk 3 above)
Comment (archit-harness) [reply pipeline-service/service/src/main/java/io/harness/pms/plan/execution/handlers/PipelineStatusUpdateEventHandler.java:127]: @hextechpal if i write another observer, we will again need to fetch status from planExecution, which we are saving the call here, what do you recommend?

Comment (hextechpal) [reply pipeline-service/service/src/main/java/io/harness/pms/plan/execution/handlers/PipelineStatusUpdateEventHandler.java:127]: I think this it is fine if you fetch the status as this query is executed just once for each pipelines. You can also look at including this in the method signature itself along with ambiance

**Comment by sahilhindwani-harness** [comment]
*[`pipeline-service/modules/orchestration/src/main/java/io/harness/engine/observers/beans/OrchestrationStartInfo.java` line 26 at r3](https://reviewable.io/reviews/harness/harness-core/51917#-NcWgbIC9q9rZj8EPSPq:-NcWgbIDDQhplTMOwVM9:bx3kpkw) ([raw file](https://github.com/harness/harness-core/blob/2317f87c1bb9b69e24a278f99005cfd6d9c80525/pipeline-service/modules/orchestration/src/main/java/io/harness/engine/observers/beans/OrchestrationStartInfo.java#L26)):*
> ```Java
>   @NotNull Ambiance ambiance;
>   @NotNull PlanExecutionMetadata planExecutionMetadata;
>   Status startStatus;
> ```

why do we need this? Since it is start, will it not always be running?

**Comment by sahilhindwani-harness** [comment]
*[`pipeline-service/modules/orchestration-visualization/src/main/java/io/harness/event/OrchestrationStartEventHandler.java` line 48 at r3](https://reviewable.io/reviews/harness/harness-core/51917#gh-1302567941:-NcWgjLR4e3PoIc10SVd:b-w16lhb) ([raw file](https://github.com/harness/harness-core/blob/2317f87c1bb9b69e24a278f99005cfd6d9c80525/pipeline-service/modules/orchestration-visualization/src/main/java/io/harness/event/OrchestrationStartEventHandler.java#L48)):*
<details><summary><i>Previously, hextechpal (Prashant Pal) wrote…</i></summary><blockquote>

Write a new observer and publish metrics there
</blockquote></details>

+1, also that observer should be async

**Review by sahilhindwani-harness — CHANGES_REQUESTED** [review]
> Reviewed 8 of 13 files at r1, 1 of 1 files at r2, 5 of 5 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51917)* status: 14 file(s) to review. 3 change requests. 2. unresolved discussions. 0 of 2 approvals obtained.

**Review by archit-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51917)* status: 32 file(s) to review. 3 change requests. 6. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/modules/orchestration/src/main/java/io/harness/engine/observers/beans/OrchestrationStartInfo.java` line 26 at r3](https://reviewable.io/reviews/harness/harness-core/51917#-NcWgbIC9q9rZj8EPSPq:-NcXGjQZ5EBHI6Vdex5r:b-65k42o) ([raw file](https://github.com/harness/harness-core/blob/2317f87c1bb9b69e24a278f99005cfd6d9c80525/pipeline-service/modules/orchestration/src/main/java/io/harness/engine/observers/beans/OrchestrationStartInfo.java#L26)):*
<details><summary><i>Previously, sahilhindwani-harness (Sahil Hindwani) wrote…</i></summary><blockquote>

why do we need this? Since it is start, will it not always be running?
</blockquote></details>

Done. it can be queued as well
___
*[`pipeline-service/service/src/main/java/io/harness/monitoring/PipelineMetricsPublisher.java` line 0 at r3](https://reviewable.io/reviews/harness/harness-core/51917#gh-1302566279:-NcXGaxr9CNry_ly5ksd:b-tfmk0z) ([raw file](https://github.com/harness/harness-core/blob/2317f87c1bb9b69e24a278f99005cfd6d9c80525/pipeline-service/service/src/main/java/io/harness/monitoring/PipelineMetricsPublisher.java#L0)):*
<details><summary><i>Previously, naidu-harness (Sanyasi Naidu Annepu) wrote…</i></summary><blockquote>

cant we remove this file altogether?
</blockquote></details>

done, will remove in next pr
___
*[`pipeline-service/service/src/main/java/io/harness/pms/plan/execution/handlers/PipelineStatusUpdateEventHandler.java` line 127 at r3](https://reviewable.io/reviews/harness/harness-core/51917#gh-1302571244:-NcXGfS54t2yd8ggrkh3:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/2317f87c1bb9b69e24a278f99005cfd6d9c80525/pipeline-service/service/src/main/java/io/harness/pms/plan/execution/handlers/PipelineStatusUpdateEventHandler.java#L127)):*
<details><summary><i>Previously, hextechpal (Prashant Pal) wrote…</i></summary><blockquote>

I think this it is fine if you fetch the status as this query is executed just once for each pipelines. You can also look at including this in the method signature itself along with ambiance
</blockquote></details>

Done.
___
*[`pipeline-service/modules/orchestration-visualization/src/main/java/io/harness/event/OrchestrationStartEventHandler.java` line 48 at r3](https://reviewable.io/reviews/harness/harness-core/51917#gh-1302567941:-NcXGg0r-s1EUPO8RD63:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/2317f87c1bb9b69e24a278f99005cfd6d9c80525/pipeline-service/modules/orchestration-visualization/src/main/java/io/harness/event/OrchestrationStartEventHandler.java#L48)):*
<details><summary><i>Previously, sahilhindwani-harness (Sahil Hindwani) wrote…</i></summary><blockquote>

+1, also that observer should be async
</blockquote></details>

Done.
___
*[`pipeline-service/service/src/main/resources/metrics/metricGroups/execution-starting-count-metric-group.yaml` line 1 at r1](https://reviewable.io/reviews/harness/harness-core/51917#gh-1302427715:-NcXGhGY5cl8jtPNRoDZ:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/59b99cac14fd95bd747be961e6e7726d85c9ec33/pipeline-service/service/src/main/resources/metrics/metricGroups/execution-starting-count-metric-group.yaml#L1)):*
<details><summary><i>Previously, Raghav-Murali wrote…</i></summary><blockquote>

Same as previous comment. No need to have separate group file, one should suffice.
</blockquote></details>

Done.

**Comment by archit-harness** [comment]
trigger uts

**Review by naidu-harness — APPROVED** [review]

**Review by sahilhindwani-harness — APPROVED** [review]
> Reviewed 20 of 22 files at r4, 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51917)* status: 3. unresolved discussions.

**Comment by archit-harness** [comment]
trigger review

**Pull request merged**